cmd/libs/go2idl/client-gen/generators/generator-for-group.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generators
+
+import (
+	"io"
+
+	"k8s.io/kubernetes/cmd/libs/go2idl/generator"
+	"k8s.io/kubernetes/cmd/libs/go2idl/namer"
+	"k8s.io/kubernetes/cmd/libs/go2idl/types"
+)
+
+// genGroup produces a file for a group client, e.g. ExtensionsClient for the extension group.
+type genGroup struct {
+	generator.DefaultGen
+	outputPackage string
+	group         string
+	// types in this group
+	types   []*types.Type
+	imports *generator.ImportTracker
+}
+
+// We only want to call GenerateType() once per group.
+func (g *genGroup) Filter(c *generator.Context, t *types.Type) bool {
+	return t == g.types[0]
+}
+
+func (g *genGroup) Namers(c *generator.Context) namer.NameSystems {
+	return namer.NameSystems{
+		"raw": namer.NewRawNamer(g.outputPackage, g.imports),
+	}
+}
+
+func (g *genGroup) Imports(c *generator.Context) (imports []string) {
+	return append(g.imports.ImportLines(), "fmt", "strings")
+}
+
+func (g *genGroup) GenerateType(c *generator.Context, t *types.Type, w io.Writer) error {
+	sw := generator.NewSnippetWriter(w, c, "$", "$")
+	m := map[string]interface{}{
+		"group":                      g.group,
+		"Group":                      namer.IC(g.group),
+		"types":                      g.types,
+		"Config":                     c.Universe.Get(types.Name{"k8s.io/kubernetes/pkg/client/unversioned", "Config"}),
+		"DefaultKubernetesUserAgent": c.Universe.Get(types.Name{"k8s.io/kubernetes/pkg/client/unversioned", "DefaultKubernetesUserAgent"}),
```

Comment:
Can you extract "k8s.io/kubernetes/pkg/client/unversioned" to a variable and use it in all places here?